flagsmith-jira-app/src/backend/flagsmith.ts

```diff
@@ -157,22 +159,59 @@ export const readEnvironments: ReadEnvironments = async ({ projectId }) => {
 };
 
 export type ReadFeatures = (args: {
-  projectId?: string;
+  projectIds?: string[];
   environmentId?: string;
 }) => Promise<Feature[]>;
 
 /** Read Flagsmith Features for stored API Key, given Project ID and optional Environment ID */
-export const readFeatures: ReadFeatures = async ({ projectId, environmentId }) => {
+export const readFeatures: ReadFeatures = async ({ projectIds, environmentId }) => {
   const apiKey = await readApiKey();
   checkApiKey(apiKey);
-  if (!projectId) throw new ApiError("Flagsmith project not connected", 400);
-  const params = new URLSearchParams({ is_archived: "false" });
-  if (environmentId) params.set("environment", environmentId);
-  const path = route`/projects/${projectId}/features/?${params}`;
-  const data = (await flagsmithApi(apiKey, path)) as PaginatedModels<Feature>;
-  const results = await unpaginate(apiKey, data);
-  if (results.length === 0) throw new ApiError("Flagsmith project has no features", 404);
-  return results;
+
+  if (!projectIds || projectIds.length === 0) {
+    throw new ApiError("Flagsmith project not connected", 400);
+  }
+
+  const allFeatures: Feature[] = [];
+  const organisationId = await readOrganisationId();
+  const projects = await readProjects({ organisationId });
+
+  for (const projectId of projectIds) {
+    const params = new URLSearchParams({ is_archived: "false" });
+    if (environmentId) params.set("environment", environmentId);
+    const path = route`/projects/${projectId}/features/?${params}`;
+
+    try {
+      const data = (await flagsmithApi(apiKey, path)) as PaginatedModels<Feature>;
+      const results = await unpaginate(apiKey, data);
+
+      const project = projects.find((p) => String(p.id) === String(projectId));
+
+      if (!project) {
+        throw new ApiError(`Flagsmith project ${projectId} not found`, 404);
+      }
```

Comment:
This check could be done at the start of the loop before the call to get features for what would be an unknown project?